### PR TITLE
Add display value of bar chart

### PR DESCRIPTION
**Pull request by dgdblank**
Allow users to optionally show the values of the bar in the bar chart.
- Includes text class for changing the styles
- Works for both horizontal and vertical bars, but only displays to the right of horizontal bars
- Includes formatting function prop for changing how the values display

### CODE REVIEW

**Comment by dandelany** [comment]
Thanks for the contribution, @dgdblank ! I haven't had time to do a full review & test yet, but just a few notes:

* For the sake of consistency with other components (eg. the axis components), I think we should use the word "label" instead of "text", eg. `labelDistance`, `labelFormat` etc. rather than `textDistance`, `textFormat`
* Also for consistency - for booleans which determine whether or not something is shown, we tend to use the prefix `show` rather than `display` (again, see axis components for example). So maybe `displayValue` -> `showLabel` (or `showBarLabel`?)
* We generally only use all-caps variable names for constants (usually strings) which are used across the library - so `RECT` and `TEXT` in `Bar.js` can just be `rect` and `text`
* Longer-term, I think we may want to implement a more generalized solution to labeling values shown on `XYPlot`s, something like a separate `XYValueLabel` or even `XYValueLabelCollection` component(s), rather than combining it into the `Bar` component. This would allow us to use the same label component for multiple types of plots, instead of just bars, and would fit more with our philosophy of composition & having each component "do one thing well". However, getting all of the props/options we'll want for generalized labels will be quite a bit of work, so I think this is a good interim solution until we figure out our long-term roadmap for value labels.

Thanks again for your work on this! Hope this doesn't seem too nitpicky but I think it's valuable for us to ensure naming consistency across the library, to reduce confusion whenever possible.

-d

File: src/Bar.js [hunk 1]
```diff
@@ -83,14 +83,31 @@ export default class Bar extends React.Component {
...
+    labelClassName: PropTypes.string
   };
   static defaultProps = {
     x: 0,
     y: 0,
     thickness: 8,
     className: "",
-    style: {}
+    style: {},
+    textDistance: 24
```
Comment (krissalvador27) [inline src/Bar.js]: is this used?

**Comment by dgdblank** [comment]
@krissalvador27 whoops, fixed.

**Comment by krissalvador27** [comment]
Like @dandelany said I agree that a long term solution around some smart HOC or something of that flavor could be implemented that can allow someone to decorate bars, scatterplot points etc in the future with provided coordinates (among other props/options). Will merge!

**Comment by dandelany** [comment]
👏 👏  thanks @krissalvador27 @dgdblank

**Pull request merged**